src/pages/settings/Profile/Contacts/NewContactMethodPage.js

```diff
@@ -52,102 +54,87 @@ const defaultProps = {
     loginList: {},
 };
 
-class NewContactMethodPage extends Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            login: '',
-            password: '',
-        };
-        this.onLoginChange = this.onLoginChange.bind(this);
-        this.validateForm = this.validateForm.bind(this);
-        this.submitForm = this.submitForm.bind(this);
-    }
-
-    onLoginChange(login) {
-        this.setState({login});
-    }
-
-    /**
-    * Determine whether the form is valid
-    *
-    * @returns {Boolean}
-    */
-    validateForm() {
-        const login = this.state.login.trim();
+function NewContactMethodPage(props) {
+    const [login, setLogin] = useState('');
+    const [password, setPassword] = useState('');
+    const loginInputRef = useRef(null);
+
+    const handleLoginChange = useCallback((value) => {
+        setLogin(value.trim());
+    }, [login]);
+
+    const handlePasswordChange = useCallback((value) => {
+        setPassword(value.trim());
+    }, [password]);
```

Comment:
I did this to fix the merge conflict caused by https://github.com/Expensify/App/pull/17817 and think it's a better solution so that the values in `state` never have leading or trailing whitespace. 

It does add one small UX quirk that I'm not sure if we care about: https://user-images.githubusercontent.com/4073354/234353895-d83bf242-44b4-48fc-8032-f85a0323c50b.mp4

basically if you go back to the start of the text and hit a `space` b/c we `.trim` the value the cursor jumps to the end of the text. 

Personally I don't really think this is a problem and probably only something QA will do while testing - real users most likely are not attempting this and will require a bit more code to prevent it but open to other opinions.